### PR TITLE
Pr/fix fullscreen

### PR DESCRIPTION
**Pull request by uli42**

### CODE REVIEW

File: nx-X11/programs/Xserver/hw/nxagent/Window.c [hunk 1]
```diff
@@ -1003,6 +1008,13 @@ void nxagentSwitchAllScreens(ScreenPtr pScreen, Bool switchOn)
       }
     }
 
+    /*
+     * FIXME: These are 0 most of the time nowadays. The effect is,
+     * that the window is moving a bit to right/bottom everytime
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Window.c]: s/everytime/every time/

File: nx-X11/programs/Xserver/hw/nxagent/Window.c [hunk 2]
```diff
@@ -1003,6 +1008,13 @@ void nxagentSwitchAllScreens(ScreenPtr pScreen, Bool switchOn)
...
+     * Fullscreen is left. To fix this ask the window manager the
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Window.c]: s/Fullscreen/fullscreen/

Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Window.c]: s/ask the window manager the frame extents/query the frame extents from the window manager/

File: nx-X11/programs/Xserver/hw/nxagent/Window.c [hunk 3]
```diff
@@ -1003,6 +1008,13 @@ void nxagentSwitchAllScreens(ScreenPtr pScreen, Bool switchOn)
...
+     * frame extents with _NET_REQUEST_FRAME_EXTENTS
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Window.c]: s/with _NET/via _NET/

File: nx-X11/programs/Xserver/hw/nxagent/Screen.c [hunk 4]
```diff
@@ -314,6 +314,28 @@ void nxagentMinimizeFromFullScreen(ScreenPtr pScreen)
   }
 }
 
+/* check if possible_parent id parent of candidate */
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Screen.c]: s/id/is/

File: nx-X11/programs/Xserver/hw/nxagent/Screen.c [hunk 5]
```diff
@@ -336,6 +336,11 @@ Bool nxagentIsParentOf(Display *d, Window possible_parent, Window candidate)
   }
 }
 
+/*
+ * In contrast to its name this function is here to take care of the
+ * iconfied window. Some window managers tend to do 'interesting'
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Screen.c]: s/iconfied/iconified/ (which window do you refer to here? The nxagent window? Please explain better)

File: nx-X11/programs/Xserver/hw/nxagent/Screen.c [hunk 6]
```diff
@@ -349,44 +354,70 @@ void nxagentMaximizeToFullScreen(ScreenPtr pScreen)
...
        We could avoid the session window is iconified when a LeaveNotify event is received,
        so this check would be unnecessary.
 */
-    struct timeval timeout;
-    int i;
-    XEvent e;
 
-    XReparentWindow(nxagentDisplay, nxagentFullscreenWindow,
-                        RootWindow(nxagentDisplay, DefaultScreen(nxagentDisplay)), 0, 0);
+    /* only reparent if necessary. FIXME: also check if the desired coordinates match*/
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Screen.c:375]: s@match*/@match */@ (blank missing near EOL)

File: nx-X11/programs/Xserver/hw/nxagent/Screen.c [hunk 7]
```diff
@@ -349,44 +354,70 @@ void nxagentMaximizeToFullScreen(ScreenPtr pScreen)
...
+        timeout.tv_sec = 0;
+        timeout.tv_usec = 50 * 1000;
+
+        nxagentWaitEvents(nxagentDisplay, &timeout);
+      }
+    }
+    else
+    {
+      #ifdef TEST
+      fprintf(stderr, "%s: FullscreenWindow already has root has parent - skipping reparenting,\n", __func__);
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Screen.c]: s/has root has parent/has root as parent/

**Review by sunweaver — CHANGES_REQUESTED** [review]
I discovered various typos in comments. Please amend.

Furthermore, why do you reference #458 as "Attributes #458" in commit message of 1787556cd43408a5269cfdd1fc2f4d0c224c3dc5? Doesn't your PR fix #458?

**Comment by uli42** [comment]
Amended. And added some more little changes. Please review, especially the explanations.

I am not 100% sure if my patch is really fixing all the problems. It works for me, but it might nor work for others. So I was a bit hesitant. Changed it...

**Review by sunweaver — APPROVED** [review]

File: nx-X11/programs/Xserver/hw/nxagent/Window.c (hunk 1 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/hw/nxagent/Window.c]: https://www.grammarly.com/blog/everytime-every-time/

File: nx-X11/programs/Xserver/hw/nxagent/Window.c (hunk 2 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/hw/nxagent/Window.c]: s/Fullscreen/fullscreen/

**Pull request merged**